server/service/apple_mdm.go

```diff
@@ -2124,23 +2124,57 @@ func (svc *Service) needsOSUpdateForDEPEnrollment(ctx context.Context, m fleet.M
 	// some cross-check against the machine info to ensure that the platform of the host aligns with
 	// what we expect from the machine info. But that would involve work to derive the platform from
 	// the machine info (presumably from the product name, but that's not a 1:1 mapping).
-	settings, err := svc.ds.GetMDMAppleOSUpdatesSettingsByHostSerial(ctx, m.Serial)
+	platform, settings, err := svc.ds.GetMDMAppleOSUpdatesSettingsByHostSerial(ctx, m.Serial)
```

Comment:
A lower-touch alternative to updating the `GetMDMAppleOSUpdatesSettingsByHostSerial` method to return the platform might be to check the `m.Product` or `m.SoftwareUpdateDeviceID` and derive the platform from that. Looks like @gillespi314 implemented this at least [one place that I can see](https://github.com/fleetdm/fleet/blob/09c2e2ff728dda1d60b3462a656be0655e965275/server/mdm/apple/gdmf/api.go#L82-L87) but I'd want a second opinion about how reliable it is.  Any thoughts Sarah?